scripts/ci/githubActions/createMatrix.ts

```diff
@@ -1,81 +1,65 @@
 /* eslint-disable no-case-declarations */
 import * as core from '@actions/core';
 
-import {
-  CLIENTS,
-  createClientName,
-  GENERATORS,
-  LANGUAGES,
-} from '../../common.js';
+import { CLIENTS, createClientName, GENERATORS, LANGUAGES } from '../../common.js';
 import {
   getClientsConfigField,
   getLanguageFolder,
   getTestExtension,
   getTestOutputFolder,
 } from '../../config.js';
-import type { Language } from '../../types.js';
-import { getNbGitDiff } from '../utils.js';
-
-import { DEPENDENCIES, COMMON_DEPENDENCIES } from './setRunVariables.js';
-import type {
-  ClientMatrix,
-  CreateMatrix,
-  Matrix,
-  SpecMatrix,
-  ToRunMatrix,
-} from './types.js';
+
+import { DEPENDENCIES } from './setRunVariables.js';
+import type { ClientMatrix, CreateMatrix, Matrix, SpecMatrix, ToRunMatrix } from './types.js';
 import { computeCacheKey, isBaseChanged } from './utils.js';
 
 // This empty matrix is required by the CI, otherwise it throws
 const EMPTY_MATRIX = { client: ['no-run'] };
 
-async function getClientMatrix(baseBranch: string): Promise<void> {
-  const matrix = LANGUAGES.reduce(
-    (curr, lang) => ({
-      ...curr,
-      [lang]: {
-        path: getLanguageFolder(lang),
-        toRun: [],
-        cacheToCompute: [],
-      },
-    }),
-    {} as Record<Language, ToRunMatrix>
-  );
+async function createClientMatrix(baseBranch: string): Promise<void> {
+  const matrix: Record<string, ToRunMatrix> = {};
```

Comment:
```suggestion
  const matrix: Record<Language, ToRunMatrix> = {};
```
this is too precise ?

Reply:
yup because the enum enforces the initialization of every keys even if it's empty :( so since we have 2 loops anyway it's good to keep a string